'Dependencies' chapter fixes

nilslindemann opened the pull request.

Commit:
Add missing word

Commit:
Typo

nilslindemann commented:
Notice, I'm just reading through the docs, will maybe add more fixes while doing so. Please don't merge immediately. Will write another message when I have read through the tutorial, ok? Then you can merge (if you like my changes ;-) )

Commit:
Reflect the class name in the code example

The linked code example ( https://github.com/tiangolo/fastapi/blob/master/docs_src/dependencies/tutorial002.py#L11 ) uses CommonQueryParams as class name, not CommonQueryParameters.

Commit:
missing dot

Kludex commented:
You can add a flag `WIP: ` on the PR title for the next PRs :)

Commit:
Simplify sentence

phy25 commented:
"Changing the stage of a pull request - GitHub Docs" https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request

Commit:
Simplify sentence

Commit:
Simplify sentence

nilslindemann commented:
Thx for the *WIP* ('Work in progress') hint, @Kludex

Commit:
Simplify sentence

Commit:
Grammar

The verb 'help' is used with an infinitive: 'to avoid' or 'avoid'.

Commit:
Typo

The verb re-use can be separated with a dash but not the adjective unused (https://en.wiktionary.org/wiki/unused). I guess that is because there is no verb "to un(-)use".

nilslindemann commented:
Ok, im done with the *dependencies* chapter.

I will create new branches for other chapters in order to not make this PR too big or prevent other people from making changes.

Can be merged (if there is interest doing so).

nilslindemann commented:
@phy25 i misread your comment yesterday (that *you* made some changes).

Thanks for this link, but i dont see such merge buttons, My own repo has no pull requests (obviously). I see [this and other PRs in this repo](https://github.com/tiangolo/fastapi/pulls), but when i click mine, i come here, there is no merge button.

Kludex commented on docs/en/docs/tutorial/dependencies/classes-as-dependencies.md:
```diff
@@ -101,7 +101,7 @@ In both cases the data will be converted, validated, documented on the OpenAPI s
 Now you can declare your dependency using this class.
 
-And as when **FastAPI** calls that class the value that will be passed as `commons` to your function will be an "instance" of the class, you can declare that parameter `commons` to be of type of the class, `CommonQueryParams`.
+**FastAPI** calls that class. This creates an "instance" of that class. The instance will be passed as `commons` to your function. You can declare that parameter `commons` to be of type of the class, `CommonQueryParams`.
```
```suggestion
**FastAPI** calls that class. This creates an "instance" of that class. Then the instance will be passed as `commons` to your function and you can declare that parameter `commons` to be of type of the class, `CommonQueryParams`.
```

nilslindemann replied:
How about

'Then the instance will be passed as parameter `commons` to your function and you can declare this parameter to be of type of the class, `CommonQueryParams`.

To be honest I dont know how to make this change, i just see a 'Commit suggestion' button, but no 'Edit' button. I fear breaking something and Git doesnt really let me undo stupid things :-)

Kludex replied:
You'll need to change locally or you can create your own suggestion and apply hahaha

Kludex replied:
But I agree with your changes :)

nilslindemann replied:
Not sure what you mean with missing part (probably because its missing ;-) )

nilslindemann replied:
Ah, ok, thanks

nilslindemann replied:
Reformulated this part a bit. Let me know if you like it.

Commit:
Reformulation

Put the sentence after the code example.

Remove the sentence 'You can declare that parameter `commons` to be of type of the class, `CommonQueryParams`.' because this is explained directly below.

Remove the horizontal ruler and replace 'In this case, ' with 'but'.

Commit:
Radical shortening of the Type annotation vs `Depends` chapter

I hope it is not too heretic :-)

This makes dependencies/tutorial003.py redundant

nilslindemann commented:
I added one more change. Let me know if this is too much.

tiangolo committed:
📝 Revert and refactor some changes for explicitness, clarity, etc.

tiangolo commented:
Great! :mag: 

I reverted a couple of changes to keep the same explicitness and clarity, etc.

Thanks for your contribution! :coffee: :cake: 

And thanks for the help @phy25 and @Kludex !
